react/features/dial-out/components/DialOutDialog.web.js

```diff
@@ -0,0 +1,233 @@
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: React.PropTypes.func
+    }
+
+    /**
+     * Initializes a new {@code DialOutNumbersForm} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            /**
+             * The number to dial.
+             */
+            dialNumber: ''
+        };
+
+        // Bind event handlers so they are only bound once for every instance.
+        this._onDialNumberChange = this._onDialNumberChange.bind(this);
+        this._onCancel = this._onCancel.bind(this);
+        this._onSubmit = this._onSubmit.bind(this);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        return (
+            <Dialog
+                okTitleKey = 'dialOut.dial'
+                onCancel = { this._onCancel }
+                onSubmit = { this._onSubmit }
+                titleKey = 'dialOut.dialOut'
+                width = 'small'>
+                { this._renderContent() }
+            </Dialog>);
+    }
+
+    /**
+     * Renders the dialog content.
+     *
+     * @returns {XML}
+     * @private
+     */
+    _renderContent() {
+        const { _isDialNumberAllowed } = this.props;
+
+        return (
+            <div className = 'dial-out-content'>
+                { _isDialNumberAllowed ? '' : this._renderErrorMessage() }
+                <DialOutNumbersForm
+                    onChange = { this._onDialNumberChange } />
+            </div>);
+    }
+
+    /**
+     * Renders the error message to display if the dial phone number is not
+     * allowed.
+     *
+     * @returns {XML}
+     * @private
+     */
+    _renderErrorMessage() {
+        const { t } = this.props;
+
+        return (
+            <div className = 'dial-out-error'>
+                { t('dialOut.phoneNotAllowed') }
+            </div>);
+    }
+
+    /**
+     * Cancel the dial out.
+     *
+     * @private
+     * @returns {boolean} - Returns true to indicate that the dialog should be
+     * closed.
+     */
+    _onCancel() {
+        this.props._cancel();
+
+        return true;
+    }
+
+    /**
+     * Dials the number.
+     *
+     * @private
+     * @returns {boolean} - Returns true to indicate that the dialog should be
+     * closed.
+     */
+    _onSubmit() {
+        const dialNumber = this.state.dialNumber;
+
+        if (this.props._isDialNumberAllowed) {
+            this.props._dial(dialNumber);
+        }
+
+        return true;
+    }
+
+    /**
+     * Updates the dialNumber and check for validity.
+     *
+     * @param {string} dialNumber - The new value of the dialNumber property.
+     * @private
+     * @returns {void}
+     */
+    _onDialNumberChange(dialNumber) {
+        this.setState({
```

Comment:
I have a concern here. I'm sorry if I'm not correct but the child DialOutNumbersForm calls this method and then this method will cause rerender (because of setState) which will create new DialOutNumbersForm component. Aren't you concerned about it?

Unfortunately I don't have good idea how to fix that currently...